addons/purchase_mrp/models/stock_move.py

```diff
@@ -20,11 +20,12 @@ def _get_price_unit(self):
             return super()._get_price_unit()
         line = self.purchase_line_id
         kit_price_unit = line._get_gross_price_unit()
+        bom_line = self.bom_line_id
         if line.currency_id != self.company_id.currency_id:
             kit_price_unit = line.currency_id._convert(kit_price_unit, self.company_id.currency_id, self.company_id, fields.Date.context_today(self), round=False)
         cost_share = self.bom_line_id._get_cost_share()
         price_unit_prec = self.env['decimal.precision'].precision_get('Product Price')
-        return float_round(kit_price_unit * cost_share * line.product_qty / self.product_qty, precision_digits=price_unit_prec)
+        return float_round(kit_price_unit * cost_share * bom_line.bom_id.product_qty / bom_line.product_qty, precision_digits=price_unit_prec)
```

Comment:
The difference between the BoM line uom and BoM uom should be something to take into account